### PR TITLE
Support the `GPTCache` caching

### PR DESCRIPTION
**Pull request by SimFG**
[GPTCache](https://github.com/zilliztech/GPTCache) is a library for creating semantic cache to store responses from LLM queries.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -83,6 +84,7 @@ pytest-watcher = "^0.2.6"
 freezegun = "^1.2.2"
 responses = "^0.22.0"
 pytest-asyncio = "^0.20.3"
+gptcache = ">=0.1.7"
```
Comment (vowelparrot) [inline pyproject.toml:87]: This should be an optional dependency since we are lazily loading the import

**Review by vowelparrot — COMMENTED** [review]
Will try to review more in-depth later. Thank you for the PR!

File: pyproject.toml (hunk 1 above)
Comment (fzliu) [reply pyproject.toml]: Fixed. Thanks for the heads-up.

**Comment by SimFG** [comment]
@hwchase17 ok, thank you so much

**Pull request closed without merging**